pkg/ccl/backupccl/restore_span_covering_test.go

```diff
@@ -82,17 +107,27 @@ func MockBackupChain(length, spans, baseFiles int, r *rand.Rand) []backuppb.Back
 // length to the number of files a file's end key was greater than any prior end
 // key when walking files in order by start key in the backups. This check is
 // thus sensitive to ordering; the coverage correctness check however is not.
+//
+// The function also verifies that a cover does not cross a span boundary.
 func checkRestoreCovering(
 	backups []backuppb.BackupManifest,
-	spans roachpb.Spans,
+	data restorationData,
 	cov []execinfrapb.RestoreSpanEntry,
 	merged bool,
 ) error {
 	var expectedPartitions int
 	required := make(map[string]*roachpb.SpanGroup)
+	spans := data.getSpans()
 	for _, s := range spans {
 		var last roachpb.Key
 		for _, b := range backups {
+			isCovered, err := data.coveredByLaterReintroduction(s.Key, b.EndTime)
```

Comment:
nit: quick comment for future readers about why we're skipping these backups, even if you just point to the comment in makeSimpleImportSpans thats fine